src/core/Chip/index.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Icon from '../Icon'
+import s from './styles'
+
+// Only allow lower case letters, optionally seperated by underscore
+const mdIconNamePattern = /^[a-z]+_*[a-z]+$/
+
+class Chip extends React.PureComponent {
+    onRemove = event => {
+        event.stopPropagation()
+        this.props.onRemove()
+    }
+
+    renderIcon(icon) {
+        return mdIconNamePattern.test(icon) ? (
+            <Icon name={icon} className={s('icon')} />
+        ) : (
+            <img src={icon} alt="chip-icon" className={s('image-icon')} />
+        )
+    }
+
+    render() {
+        const {
+            label,
+            onClick,
+            selected,
+            disabled,
+            dragging,
+            icon,
+            onRemove,
+        } = this.props
+
+        return (
+            <div
+                className={s('container', {
+                    static: !onClick,
+                    selected,
+                    disabled,
+                    dragging,
+                })}
+                onClick={(!disabled && onClick) || undefined}
+            >
+                {icon && this.renderIcon(icon)}
+                {label && <span className={s('label')}>{label}</span>}
+                {onRemove && (
+                    <Icon
+                        name="cancel"
+                        className={s('remove-icon')}
+                        onClick={this.onRemove}
+                    />
+                )}
+            </div>
+        )
+    }
+}
+
+Chip.propTypes = {
```

Comment:
- add type property to tackle `icon`, `image`
- add optional `src` in case Icon on the side is a image

Reply:
If we add the `src` prop couldn't we infer that it's an `image`?

then we could just use: `icon` or ``src` as props and drop `type`.

```
<Chip src="<url>" />
<Chip icon="<icon name>" />
```

Reply:
i think `type` and `icon` read nicely and leave it flexible enough for other types without changing the api, no need for `src`:

```
<Chip icon="material_icon_1" />
<Chip type="image" icon="/img_icon.png" />
<Chip type="glyph" icon="\xE2\x9C\x8F" />
```